File: src/com/kingmang/lazurite/parser/pars/Token.java
package com.kingmang.lazurite.parser.pars;


public final class Token {

    private final TokenType type;
    private final String text;
    private final int row, col;
    
    public Token(TokenType type, String text, int row, int col) {
        this.type = type;
        this.text = text;
        this.row = row;
        this.col = col;
    }

    public TokenType getType() {
        return type;
    }

    public String getText() {
        return text;
    }

    public int getRow() {
        return row;
    }

    public int getCol() {
        return col;
    }
    
    public String position() {
        return "[" + row + " " + col + "]";
    }

    @Override
    public String toString() {
        return type.name() + " " + position() + " " + text;
    }
}

File: src/main/java/com/kingmang/lazurite/parser/tokens/Token.kt
package com.kingmang.lazurite.parser.tokens

import com.kingmang.lazurite.parser.IToken


data class Token(
    val type: TokenType,
    val text: String,
    val row: Int,
    val col: Int
) : IToken{
    override fun getType(): TokenType {
        return type
    }

    override fun getText(): String {
        return text
    }

    override fun getCol(): Int {
        return col
    }

    override fun getRow(): Int {
        return row
    }

    override fun toString(): String {
        return "${type.name}[$row, $col]" + text.takeIf { it.isNotBlank() }?.let { " \"$it\"" }.orEmpty()
    }
}
